# Applying SVM algorithm on Titanic Dataset

In [1]:
import pandas as pd
from sklearn import preprocessing
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix

In [2]:
titanic = pd.read_csv('train.csv')
titanic.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [3]:
le = preprocessing.LabelEncoder()
le.fit(titanic['Sex'])
titanic['Sex']= le.transform(titanic['Sex'])
le.fit(titanic['Embarked'])
titanic['Embarked']= le.transform(titanic['Embarked'])
le.fit(titanic['Age'])
titanic['Age']= le.transform(titanic['Age'])
le.fit(titanic['Fare'])
titanic['Fare']= le.transform(titanic['Fare'])
titanic.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex  Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    1   28      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0   51      1      0   
2                             Heikkinen, Miss. Laina    0   34      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0   47      1      0   
4                           Allen, Mr. William Henry    1   47      0      0   

             Ticket  Fare Cabin  Embarked  
0         A/5 21171    18   NaN         2  
1          PC 17599   207   C85         0  
2  STON/O2. 3101282    41   NaN         2  
3            113803   189  C123         2  
4            373450    43   NaN         2

In [4]:
from sklearn import svm

# Accuracy score and confusion matrix for Pclass as dependent variable

In [5]:
DV = ['Survived','Pclass','Sex','SibSp','Parch','Embarked']
for i in DV:
    y = titanic[i]
    X = titanic.drop([i,'PassengerId','Name','Ticket','Cabin'], axis = 1)
    X_train,X_test,y_train,y_test = train_test_split(X, y , test_size=0.3, random_state=0)
    clf = svm.SVC(gamma = 0.01, C = 100)
    clf.fit(X_train, y_train)
    y_pred = clf.predict(X_test)
    a = accuracy_score(y_test, y_pred, normalize=True)
    b = confusion_matrix(y_test, y_pred)
    print("accuracy score for", i ,"is: \n", a)
    print("Confusion Matrix for", i ,"is: \n", b)

accuracy score for Survived is: 
 0.6928838951310862
Confusion Matrix for Survived is: 
 [[129  28]
 [ 54  56]]
accuracy score for Pclass is: 
 0.8876404494382022
Confusion Matrix for Pclass is: 
 [[ 59   8   3]
 [  5  40   4]
 [  2   8 138]]
accuracy score for Sex is: 
 0.6591760299625468
Confusion Matrix for Sex is: 
 [[ 56  42]
 [ 49 120]]
accuracy score for SibSp is: 
 0.7378277153558053
Confusion Matrix for SibSp is: 
 [[151  25   4   0   2   0   0]
 [ 21  37   6   1   0   0   0]
 [  5   2   1   0   0   0   0]
 [  0   3   1   3   0   0   0]
 [  0   0   0   0   2   0   0]
 [  0   0   0   0   0   1   0]
 [  0   0   0   0   0   0   2]]
accuracy score for Parch is: 
 0.7715355805243446
Confusion Matrix for Parch is: 
 [[178  14   6   0   0   0]
 [ 20  18   1   0   0   1]
 [  9   8  10   0   0   0]
 [  0   0   1   0   0   0]
 [  0   1   0   0   0   0]
 [  0   0   0   0   0   0]]
accuracy score for Embarked is: 
 0.7827715355805244
Confusion Matrix for Embarked is: 
 [[ 31   0  18]
 [  

# From above output it is clear that When Dependent var is Pclass, it has higher accuracy with 88.76%